### PR TITLE
24167 Handle mapping of the SCS Product Updated event to catalogue notification

### PR DESCRIPTION
**Pull request by ganpat-tech**
PR contain  code changes, unit test cases and Function test cases.

### CODE REVIEW

**Review by usha13529 — CHANGES_REQUESTED** [review]
Few minor comments to fix please.

File: UKHO.ExternalNotificationService.API/UKHO.ExternalNotificationService.API.FunctionalTests/D365PayloadTestData/D365ScsSamplePayload.json [hunk 1]
```diff
@@ -0,0 +1,354 @@
...
+                    {
+                        "key": "ukho_subscriptiontype",
+                        "value": "ADDS Data Pipeline"
+                    },
+                    {
+                        "key": "ukho_subscriptionstate",
+                        "value": "Test"
+                    }
+                ],
+                "Id": "bd0ccbc8-2975-ec11-8943-002248818243",
```
[the remark this answers is not in the document]
Comment (vipin14904) [reply UKHO.ExternalNotificationService.API/UKHO.ExternalNotificationService.API.FunctionalTests/D365PayloadTestData/D365ScsSamplePayload.json]: Changes done

File: UKHO.ExternalNotificationService.API/UKHO.ExternalNotificationService.API.FunctionalTests/FunctionalTests/EnsEnterpriseEventServiceWebhookTest.cs [hunk 2]
```diff
@@ -20,6 +21,8 @@ class EnsEnterpriseEventServiceWebhookTest
         private string EnsToken { get; set; }
         private JObject FssEventBody { get; set; }
         private FssEventData FssEventData { get; set; }
+        private JObject ScsEventBody { get; set; }
+        private ScsEventData ScsEventData { get; set; }
```
[the remark this answers is not in the document]
Comment (vipin14904) [reply UKHO.ExternalNotificationService.API/UKHO.ExternalNotificationService.API.FunctionalTests/FunctionalTests/EnsEnterpriseEventServiceWebhookTest.cs]: Removed

File: UKHO.ExternalNotificationService.API/UKHO.ExternalNotificationService.API.FunctionalTests/Helper/ScsEventDataBase.cs [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+            ensWebhookJson["Source"] = $"{testConfigure.ScsSource}";
+            ensWebhookJson["Id"] = "0d2f05f5-3691-476a-9011-6007bcaa9cbf";
+            ensWebhookJson["Subject"] = "NO4F1617";
+            ensWebhookJson["DataContentType"] = "application/json";
+            ensWebhookJson["Data"] = JObject.FromObject(GetScsEventData());
+
+            return ensWebhookJson;
+        }
+
+        public static ScsEventData GetScsEventData()
```
[the remark this answers is not in the document]
Comment (vipin14904) [reply UKHO.ExternalNotificationService.API/UKHO.ExternalNotificationService.API.FunctionalTests/Helper/ScsEventDataBase.cs]: Done

**Review by usha13529 — APPROVED** [review]
Thanks for the changes. LGTM.

File: UKHO.ExternalNotificationService.API/UKHO.ExternalNotificationService.API.FunctionalTests/FunctionalTests/EnsEnterpriseEventServiceWebhookTest.cs [hunk 4]
```diff
@@ -172,5 +175,41 @@ public async Task WhenICallTheEnsWebhookApiWithAValidJObjectBody_ThenNonOkStatus
             Assert.NotNull(getMatchingData);
             Assert.Greater(getMatchingData.Count(), 1);
         }
+
+        [Test]
+        public async Task WhenICallTheEnsWebhookApiWithAValidScsJObjectBody_ThenOkStatusIsReturned()
```
[the remark this answers is not in the document]
Comment (ganpat-tech) [reply UKHO.ExternalNotificationService.API/UKHO.ExternalNotificationService.API.FunctionalTests/FunctionalTests/EnsEnterpriseEventServiceWebhookTest.cs:180]: Thanks For highlighting , done changes.

**Review by barrie-cooper — APPROVED** [review]
LGTM

**Pull request merged**